src/services/compute-event-and-metrics/get-recent-deployments.ts

```diff
@@ -1,17 +1,58 @@
 import { DataProviderDeploymentEvent } from '@atlassian/forge-graphql';
 
-import { Deployment, Environment, EnvironmentTier } from '../../types';
+import { Deployment, EnvironmentTier } from '../../types';
 import { getRecentDeployments, gitlabAPiDeploymentToCompassDataProviderDeploymentEvent } from '../deployment';
 import { getProjectEnvironments } from '../environment';
 import { getDateInThePast } from '../../utils/time-utils';
+import { isSendStagingEventsEnabled } from '../feature-flags';
 
-export const getDeploymentsForProductionEnvironments = async (
+const newGetDeploymentsForEnvironments = async (
   groupToken: string,
   projectId: number,
   projectName: string,
-  projectEnvironments?: Environment[],
+  environmentTiers: EnvironmentTier[],
 ): Promise<DataProviderDeploymentEvent[]> => {
-  const environments = projectEnvironments || (await getProjectEnvironments(projectId, groupToken));
+  // get all project environments
+  const projectEnvironments = await getProjectEnvironments(projectId, groupToken);
+
+  const deploymentEventsByEnvironmentTier = environmentTiers.map(async (environmentTier) => {
+    // get a list of deployments for each project environment belonging to the environmentTier
+    const getDeploymentsPromises = projectEnvironments.reduce<Promise<Deployment[]>[]>(
```

Comment:
My slight preference here is to do something like 
`const getDeploymentsPromises = await Promise.all(projectEnvironments.filter((env) => env.tier in environmentTiers).flatMap((env) => getRecentDeployments(groupToken, projectId, getDateInThePast(), env.name)))`

Reply:
I've adjusted it to use `.filter(...).map(...)` instead